src/main/webapp/app/exam/manage/student-exams/student-exam-detail.component.ts

```diff
@@ -118,8 +124,13 @@ export class StudentExamDetailComponent implements OnInit {
                 exercise.studentParticipations?.length &&
                 exercise.studentParticipations.length > 0 &&
                 exercise.studentParticipations[0].results?.length &&
-                exercise.studentParticipations[0].results.length > 0
+                exercise.studentParticipations[0].results!.length > 0
             ) {
+                if (exercise!.studentParticipations[0].submissions && exercise!.studentParticipations[0].submissions!.length > 0) {
+                    exercise!.studentParticipations[0].submissions![0].results! = exercise.studentParticipations[0].results;
+                    setLatestSubmissionResult(exercise?.studentParticipations[0].submissions?.[0], getLatestSubmissionResult(exercise?.studentParticipations[0].submissions?.[0]));
```

Comment:
Putting submissions into a variable might reduce the repetition here

Reply:
yes but then we lose the exercise relation to the submissions. This could be solves via a map, but this would mean a function call of get() to retrieve any value, and additionally for quiz exercises are handled differently..

Reply:
Done via new component!